examples/nffPktgen/testing/minPktLoss.go

```diff
@@ -0,0 +1,210 @@
+                time.Sleep(time.Duration(trafDelay) * time.Second)
+                pktRX := portStats.packetCount
+                pktTX := gen.GetGeneratedNumber()
+                pktLoss := calcPktLoss(pktRX, pktTX)
+                totalPktRX += pktRX
+                totalPktTX += pktTX
+                printStats(&portStats, pktRX, pktTX, pktLoss, started, currSpeed, float64(speed))
+
+                // binary search:
+                for low <= high {
+                        mid := (low + high) / 2
+                        currSpeed = mid
+                        updatedSpeed := 0.01 * float64(uint64(currSpeed) * speed)
+
+                        // reset counters:
+                        portStats.packetCount = 0
+                        gen.Count = 0
+
+                        genChan <- uint64(updatedSpeed)
+                        time.Sleep(time.Duration(trafDelay) * time.Second)
+
+                        pktRX = portStats.packetCount
+                        pktTX = gen.GetGeneratedNumber()
+                        pktLoss = calcPktLoss(pktRX, pktTX)
+
+                        if pktLoss <= tgtLoss {
+                                low = mid + 1 // tgt met so try higher speed
+                                if currSpeed > maxSpeed {
+                                        maxSpeed = currSpeed
+                                        maxSpeedPkt = updatedSpeed
+                                }
+                        } else {
+                                high = mid - 1 // tgt failed so try lower speed
+                        }
+
+                        totalPktRX += pktRX
+                        totalPktTX += pktTX
+                        // fmt.Printf("\nmid: %d\nupdatedSpeed: %d%%(%f pkt/s)\npktCountTX: %d\npktCountRX: %d\npktLoss: %f\nlow: %d\nhigh: %d\n\n", mid, currSpeed, updatedSpeed, pktTX, pktRX, pktLoss, low, high) // debugging purposes
+
+                        printStats(&portStats, pktRX, pktTX, pktLoss, started, currSpeed, updatedSpeed)
+                }
+                fmt.Printf("--------------------BINARY SEARCH COMPLETE!!!--------------------\n")
+                interruptChannel <- os.Interrupt
+        }()
+
+
+out:
+        for {
+                select {
+                case sig := <-interruptChannel:
+                        fmt.Printf("Received signal %v, finishing.\n", sig)
+                        break out
+                case <-finishChannel:
+                        fmt.Println("Test timeout reached")
+                        break out
+                case <-statsChannel:
+                        printStats(&portStats, totalPktRX, totalPktTX, calcPktLoss(totalPktRX, totalPktTX), started, maxSpeed, maxSpeedPkt)
+                }
+        }
+        printTotals(&portStats, totalPktRX, totalPktTX, started, maxSpeed, maxSpeedPkt)
+}
+
+func calcPktLoss(pktCountRX uint64, pktCountTX uint64) float64 {
+        return (float64(pktCountTX - pktCountRX) / float64(pktCountTX))
```

Comment:
You can add `* 100` here and then everywhere in program your fractions will switch to percents automatically :)